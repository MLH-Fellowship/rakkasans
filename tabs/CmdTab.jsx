import React, { useState } from 'react';
import {
  View,
  StyleSheet,
  SafeAreaView,
  Modal,
  Text,
} from 'react-native';
import { useNavigation } from '@react-navigation/native';

import Colors from '../constants/Colors';
import CommandCard from '../components/CommandCard';
import SquareButton from '../components/SquareButton';
import RectButton from '../components/RectButton';
import Dimensions from '../constants/Dimensions';

import Copic from '../assets/images/COPIC.jpeg';
import SgmPic from '../assets/images/SGMPIC.jpeg';

export default function CmdTab() {
  const navigation = useNavigation();
  const [modal1Visible, setModal1Visible] = useState(false);
  const [modal2Visible, setModal2Visible] = useState(false);

  const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: Colors.white,
      width: '100%',
      alignItems: 'center',
      justifyContent: 'center',
    },
    cmdTeamContainer: {
      justifyContent: 'center',
      width: '100%',
      marginTop: 30,
    },
    buttonTopPair: {
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'space-evenly',
      width: '100%',
    },
    buttonBottomPair: {
      marginTop: 35,
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'space-evenly',
      width: '100%',
    },
    buttonsView: {
      alignItems: 'center',
      justifyContent: 'center',
      flex: 1,
    },
    modalView: {
      margin: 20,
      width: '90%',
      height: '90%',
      backgroundColor: 'white',
      borderRadius: 20,
      padding: 35,
      alignItems: 'center',
      justifyContent: 'space-between',
      shadowColor: '#000',
      shadowOffset: {
        width: 0,
        height: 2,
      },
      shadowOpacity: 0.6,
      shadowRadius: 3.84,
      elevation: 8,
    },
    modalTitle: {
      fontSize: 20,
      color: Colors.darkGray,
      marginBottom: 10,
    },
    modalBody: {
      fontSize: 16,
      color: Colors.darkGray,
    },
    modalBio: {
      alignItems: 'center',
    },
    centeredView: {
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',

      // marginTop: 22,
    },
  });
  return (
    <SafeAreaView style={styles.container}>
      <Modal animationType="slide" transparent visible={modal1Visible}>
        <View style={styles.centeredView}>
          <View style={styles.modalView}>
            <View style={styles.modalBio}>
              <Text style={[Dimensions.font, styles.modalTitle]}>Bio #1</Text>
              <Text style={[Dimensions.font, styles.modalBody]}>
                Here is a sample bio with a fair amount of text which should
                span multiple lines
              </Text>
            </View>
            <RectButton
              text="Dismiss"
              onPress={() => {
                setModal1Visible(!modal1Visible);
              }}
            />
          </View>
        </View>
      </Modal>

      <Modal animationType="slide" transparent visible={modal2Visible}>
        <View style={styles.centeredView}>
          <View style={styles.modalView}>
            <View style={styles.modalBio}>
              <Text style={[Dimensions.font, styles.modalTitle]}>Bio #2</Text>
              <Text style={[Dimensions.font, styles.modalBody]}>
                Here is a sample bio with a fair amount of text which should
                span multiple lines
              </Text>
            </View>
            <RectButton
              text="Dismiss"
              onPress={() => {
                setModal2Visible(!modal2Visible);
              }}
            />
          </View>
        </View>
      </Modal>

      <View style={styles.cmdTeamContainer}>
        <CommandCard
          title="COL Brandon Teague"
          source={Copic}
          onPress={() => {
            setModal1Visible(true);
          }}
        />
        <CommandCard
          title="CSM Eddie Brewer"
          source={SgmPic}
          onPress={() => {
            setModal2Visible(true);
          }}
        />
      </View>
      <View style={styles.buttonsView}>
        <View style={[styles.buttonTopPair, { paddingTop: 20 }]}>
          <SquareButton
            name="email-open"
            text="Welcome Letter"
            buttonSize={90}
            textSize={12}
            onPress={() => navigation.navigate('Welcome Letter')}
          />
          <SquareButton
            name="eye-circle"
            text="Vision" // Could change to "Commanders' Vision", but it doesn't fit
            buttonSize={90}
            textSize={13}
            onPress={() => navigation.navigate("Commanders' Vision")}
          />
        </View>
        <View style={[styles.buttonBottomPair, { paddingBottom: 25 }]}>
          <SquareButton
            name="email-multiple"
            text="Policy Letters"
            buttonSize={90}
            textSize={12}
            onPress={() => navigation.navigate('Policy Letters')}
          />
          <SquareButton
            name="home-city"
            text="Off Limits" // Could change to "Off Limit Establishments", but it doesn't fit
            buttonSize={90}
            textSize={12}
            onPress={() => navigation.navigate('Off Limits')}
          />
        </View>
      </View>
    </SafeAreaView>
  );
}
