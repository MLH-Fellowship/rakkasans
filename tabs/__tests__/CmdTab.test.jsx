import React from 'react';
import renderer from 'react-test-renderer';

import CmdTab from '../CmdTab';

jest.useFakeTimers();
jest.mock('@react-navigation/native');

it('renders correctly', () => {
  const tree = renderer.create(<CmdTab />).toJSON();
  expect(tree).toMatchSnapshot();
});
