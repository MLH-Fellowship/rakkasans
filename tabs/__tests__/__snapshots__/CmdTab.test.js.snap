// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flex": 1,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 20,
            "elevation": 8,
            "height": "90%",
            "justifyContent": "space-between",
            "margin": 20,
            "padding": 35,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.6,
            "shadowRadius": 3.84,
            "width": "90%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "andale-mono",
                },
                Object {
                  "color": "#333333",
                  "fontSize": 20,
                  "marginBottom": 10,
                },
              ]
            }
          >
            Bio #1
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "andale-mono",
                },
                Object {
                  "color": "#333333",
                  "fontSize": 16,
                },
              ]
            }
          >
            Here is a sample bio with a fair amount of text which should span multiple lines
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#020a41",
              "borderRadius": 5,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginVertical": 10,
              "opacity": 1,
              "padding": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#ffffff",
                  "fontSize": 18,
                  "marginHorizontal": 30,
                  "marginTop": 7,
                },
                Object {
                  "fontFamily": "andale-mono",
                },
              ]
            }
          >
            Dismiss
          </Text>
        </View>
      </View>
    </View>
  </Modal>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 20,
            "elevation": 8,
            "height": "90%",
            "justifyContent": "space-between",
            "margin": 20,
            "padding": 35,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.6,
            "shadowRadius": 3.84,
            "width": "90%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "andale-mono",
                },
                Object {
                  "color": "#333333",
                  "fontSize": 20,
                  "marginBottom": 10,
                },
              ]
            }
          >
            Bio #2
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "andale-mono",
                },
                Object {
                  "color": "#333333",
                  "fontSize": 16,
                },
              ]
            }
          >
            Here is a sample bio with a fair amount of text which should span multiple lines
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#020a41",
              "borderRadius": 5,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginVertical": 10,
              "opacity": 1,
              "padding": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#ffffff",
                  "fontSize": 18,
                  "marginHorizontal": 30,
                  "marginTop": 7,
                },
                Object {
                  "fontFamily": "andale-mono",
                },
              ]
            }
          >
            Dismiss
          </Text>
        </View>
      </View>
    </View>
  </Modal>
  <Image
    source={1}
    style={
      Object {
        "height": 30,
        "paddingBottom": 10,
        "top": "2%",
        "width": 30,
      }
    }
  />
  <View
    style={
      Object {
        "justifyContent": "center",
        "marginTop": 30,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderColor": "#020a41",
          "borderRadius": 5,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginHorizontal": 20,
          "marginTop": 35,
          "opacity": 1,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={1}
        style={
          Object {
            "borderBottomLeftRadius": 5,
            "borderTopLeftRadius": 5,
            "height": 100,
            "marginRight": 20,
            "width": 75,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "fontSize": 20,
              },
              Object {
                "fontFamily": "andale-mono",
              },
            ]
          }
        >
          COL Brandon Teague
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderColor": "#020a41",
          "borderRadius": 5,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginHorizontal": 20,
          "marginTop": 35,
          "opacity": 1,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={1}
        style={
          Object {
            "borderBottomLeftRadius": 5,
            "borderTopLeftRadius": 5,
            "height": 100,
            "marginRight": 20,
            "width": 75,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "fontSize": 20,
              },
              Object {
                "fontFamily": "andale-mono",
              },
            ]
          }
        >
          CSM Eddie Brewer
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-evenly",
            "width": "100%",
          },
          Object {
            "paddingTop": 20,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderRadius": 15,
            "elevation": 5,
            "opacity": 1,
            "padding": 10,
            "shadowColor": "#a0a0a0",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 8,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "height": 90,
                "width": 90,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#333333",
                  "paddingBottom": 6,
                  "textAlign": "center",
                },
                Object {
                  "fontFamily": "andale-mono",
                },
                Object {
                  "fontSize": 12,
                },
              ]
            }
          >
            Welcome Letter
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 25,
                "height": 50,
                "justifyContent": "center",
                "width": 50,
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderRadius": 15,
            "elevation": 5,
            "opacity": 1,
            "padding": 10,
            "shadowColor": "#a0a0a0",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 8,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "height": 90,
                "width": 90,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#333333",
                  "paddingBottom": 6,
                  "textAlign": "center",
                },
                Object {
                  "fontFamily": "andale-mono",
                },
                Object {
                  "fontSize": 13,
                },
              ]
            }
          >
            Vision
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 25,
                "height": 50,
                "justifyContent": "center",
                "width": 50,
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-evenly",
            "marginTop": 35,
            "width": "100%",
          },
          Object {
            "paddingBottom": 25,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderRadius": 15,
            "elevation": 5,
            "opacity": 1,
            "padding": 10,
            "shadowColor": "#a0a0a0",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 8,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "height": 90,
                "width": 90,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#333333",
                  "paddingBottom": 6,
                  "textAlign": "center",
                },
                Object {
                  "fontFamily": "andale-mono",
                },
                Object {
                  "fontSize": 12,
                },
              ]
            }
          >
            Policy Letters
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 25,
                "height": 50,
                "justifyContent": "center",
                "width": 50,
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderRadius": 15,
            "elevation": 5,
            "opacity": 1,
            "padding": 10,
            "shadowColor": "#a0a0a0",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 8,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "height": 90,
                "width": 90,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#333333",
                  "paddingBottom": 6,
                  "textAlign": "center",
                },
                Object {
                  "fontFamily": "andale-mono",
                },
                Object {
                  "fontSize": 12,
                },
              ]
            }
          >
            Off Limits
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 25,
                "height": 50,
                "justifyContent": "center",
                "width": 50,
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
